File: lecture6/readability.py
# get_string and print, with multiple arguments
from cs50 import get_string

# def main():
text = get_string("text: ")  # get input from user
letter = 0  # number of letters
word = 0  # number of words
sen = 0  # number of sentences
t = len(text)  # change to array
for i in range(t):  # for loop. loop in each element in the array
    if text[i].isalnum() == True:  # if it is a letter
        letter += 1  # add one to letter for count

for i in range(t):  # for loop. loop in each element in the array
    if text[i].isspace() == True and text[i+1].isalnum() == True:  # if (text[i] == ' ') or (text[i + 1].isalnum()):
        word += 1  # add one to word for count

for i in range(t):  # for loop. loop in each element in the array
    if text[i] == "?" or text[i] == "." or text[i] == "!":  # if (text[i] == '?') or (text[i] == '.') or (text[i] == '!'):
        sen += 1  # add one to sen for count

L = (letter / word) * 100  # calculation for letter per 100 words
S = (sen / word) * 100  # calculation for sentences per 100 words
grade = (0.0588 * L) - (0.296 * S) - 15.8  # calculation for determination which grade level

if grade < 1:  # if grade is less than 1
    print("Before Grade 1")  # print
elif grade < 16:  # if grade is less than 16
    print("Grade", round(grade))  # print
else:  # else
    print("Grade 16+")  # print

# if __name__ == "__main__":
 #   main()

File: lecture2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h> // for the length of the string 
#include <ctype.h> // for the conditons to determine the words 
#include <math.h> // for rounding and calculaitons for the level of grade in the text 

int main(void)
{
    string text = get_string("enter text: \n");//enter text by user
    float letter = 0, word = 1, sen = 0; // count for letter,word and sentence
    for (int i = 0; i < strlen(text); i++) // the length of the text 
    { 
        // if statements for counting the variables 
        if ((text[i] == ' ' || text[i] == '"' || text[i] == ':') && (isalpha(text[i + 1]))) 
        {
            word++; // +1 to word 
        }
        else  if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sen++; // +1 to sentence
        }
        else if ((text[i] <= 'z' && text[i] >= 'a') || (text[i] <= 'Z' && text[i] >= 'A'))
        {
            letter++; // +1 to letter 
        }
    } 
    //calculations 
    float L = (letter / word) * 100;
    float S = (sen / word) * 100;
    float index = (0.0588 * L) - (0.296 * S) - 15.8; //Coleman-Liau index
    // printf("%.2f letter,%.2f words, %.2f sentences, %.2f index",letter, word, sen, index); 
    //printf("%i\n", (int)round(index));
    // different conditions for determining the grade level of the text 
    if (index >= 16) // if index is greater or equal to 16 
    {
        printf("Grade 16+\n"); // print grade 16+
    }
    else if (index < 1) //if index is smaller than 1
    {
        printf("Before Grade 1\n"); // print before grade 1 
    }
    else if (index <= 15 && index > 0) // if index is greater than 0 and smaller or equal to 15
    {
        printf("Grade %i\n", (int) round(index));// cast float to int for a whole number 
        // print the grade level 
    }
}


